zebrad/src/components/mempool/error.rs

```diff
@@ -0,0 +1,29 @@
+//! Errors that can occur when adding transactions to the mempool or querying it.
+
+use thiserror::Error;
+
+#[derive(Error, Debug, PartialEq)]
+#[allow(dead_code)]
+pub enum TransactionInsertError {
+    #[error("transaction already exists in mempool")]
+    InMempool,
+
+    #[error("transaction did not pass consensus validation")]
+    TransactionInvalid(#[from] zebra_consensus::error::TransactionError),
+
+    #[error("transaction is committed in block {0:?}")]
+    TransactionInBlock(zebra_chain::block::Hash),
+
+    #[error("transaction has expired")]
+    TransactionExpired,
+
+    #[error("transaction fee is too low for the current mempool state")]
+    LowFee,
+}
+
+#[derive(Error, Debug, PartialEq)]
+#[allow(dead_code)]
+pub enum TransactionQueryError {
+    #[error("transaction was not found in mempool")]
+    NotInMempool,
+}
```

Comment:
I wonder if this is an error. This can definitely happen, but it seems to be part of normal behavior, vs a failure, so in such a case the query would return 'false' or empty set when looking for a transaction in the mempool.

Reply:
Maybe Blocker: This isn't an error in Zebra's current implementation:
https://github.com/ZcashFoundation/zebra/blob/bc4194fcb93e41b9482acb2a6626933c950d5695/zebra-network/src/protocol/internal/response.rs#L17-L26

The first table in this document says how Zebra responds to each inbound mempool message:
https://docs.google.com/document/d/1KgPl3r_Pp8AUM0KvDnBv7LK0vRMvxVjM9ooIwr2mB7U/edit#

Reply:
I removed `TransactionQueryError` then.